fusillade-api.yml

```diff
@@ -4,164 +4,465 @@ info:
+      responses:
+        200:
+          description: The result of the policy evaluation
+          content:
+            application/json:
+              schema:
+                decision:
+                  type: boolean
+        default:
+          description: Unexpected error
+  /users/{user_id}:
+    get:
+      summary: Retrieve information about a user
+      security:
+        - bearerAuth: []
       parameters:
-        - name: json_request_body
-          in: body
-          required: true
-          schema:
-            type: object
-            properties:
-              principal:
-                description: Attested user identifier
-                type: string
-              action:
-                description: The action the principal is attempting to perform
-                type: string
-              resource:
-                description: The resource the principal will perform the action against
-                type: string
-      response:
-        200:
-          description: the principal is authorized
-        401:
-          description: the principal is not authorized
+        - $ref: '#/components/parameters/user_id'
+      responses:
+        200:
+          description: Information about the user is returned
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        404:
+          description: Resource not found.
         default:
           description: Unexpected error
-  /users:
     put:
-      summary: Create or update a user and attached policy.
+      summary: Create a new user
       security:
-        - openId:
-            - https://www.googleapis.com/auth/userinfo.email
+        - bearerAuth: []
       parameters:
-        - name: json_request_body
-          in: body
-          required: true
-          schema:
-            type: object
-            properties:
-              policy:
-                description: IAM policy to attach to the user, if any
-                type: object
+        - $ref: '#/components/parameters/user_id'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                groups:
+                  type: array
+                  items:
+                    type: string
+                roles:
+                  type: array
+                  items:
+                    type: string
+                policy:
+                  type: string
       responses:
-        201:
-          description: OK
-          schema:
-            type: object
+        200:
+          description: User was created
+        401:
+          description: The user is not authorized because the object has already been created and the user does not have permission to modify, or user does not have permission to create objects.
         default:
           description: Unexpected error
-  /users/{user_id}:
+  /users/{user_id}/groups:
```

Comment:
im assuming possibility for user removal from a group at a later time?

Reply:
you could use the same API to remove a user.